ExceptionsCases.cs

```diff
@@ -0,0 +1,202 @@
+                log.Info(String.Format("Attempting to divide {0} by {1}", x, y));
+                Debug.WriteLine(int.Parse(x) / int.Parse(y));
+            }
+            catch (SystemException ex)
+            {
+                switch (ex)
+                {
+                    case FormatException f:
+                        Debug.WriteLine($"An Exception has occured: {f}");
+                        log.Warn(String.Format("Exception is caught: {0}", f));
+                        break;
+                    case DivideByZeroException d:
+                        Debug.WriteLine($"An Exception has occured: {d}");
+                        log.Warn(String.Format("Exception is caught: {0}", d));
+                        break;
+                    case OverflowException o:
+                        Debug.WriteLine($"An Exception has occured: {o}");
+                        log.Warn(String.Format("Exception is caught: {0}", o));
+                        break;
+                }
+            }
+
+        }
+
+        public void DivideIntegerWithSwitchPattern(string x, string y)
+        {
+            try
+            {
+                log.Info(String.Format("Attempting to divide {0} by {1}", x, y));
+                Debug.WriteLine(int.Parse(x) / int.Parse(y));
+            }
+            catch (SystemException ex)
+            {
+                string str = ex switch
+                {
+                    FormatException => "Format Exception!",
+                    DivideByZeroException => "Divide By Zero Exception!",
+                    OverflowException => "Overflow Exception!",
+                    _ => "Unknown Exception!"
+                };
+                Debug.WriteLine(str);
+                log.Warn(String.Format("Exception is caught: {0}", ex));
+
+            }
+        }
+
+        public void FindItemInCollection(int x)
+        {
+            List<Pet> pets = new List<Pet>();
+            pets.Add(new Pet("Hamster", "Koko", 1));
+            pets.Add(new Pet("Dog", "Chester", 14));
+            pets.Add(new Pet("Cat", "Masha", 5));
+
+            try
+            {
+                log.Info(String.Format("Looking for pet with weight = {0}", x));
+                Pet found = pets.Find(item => item.Weight == x);
+                Debug.WriteLine($"{ found.Type}, { found.Alias}");
+            }
+            catch (SystemException n)
+            {
+                Debug.WriteLine($"An Exception has occured: {n}");
+                log.Error(String.Format("Exception is thrown: {0}", n));
+                throw new Exception($"The given item was not found");
+            }
+
+        }
+
+        public void CreateFile()
+        {
+            string path = @"C:\Users\obartkiv\source\repos\HelloWorld\FileC.txt";
+            string message = "What a beautiful day!";
+
+            try
+            {
+                log.Info(String.Format("Creating a file {0}", path));
+                FileStream fs = File.Create(path);
```

Comment:
think of better method logic. atm you're creating file first and then check if filename is in the correct format